### ISSUE TITLE
Training and Validation

### ISSUE THREAD

**Issue by aarushis18** [MichiganDataScienceTeam/F24-mini-copilot#16]
File dedicated to training, validating, and saving the model. 

(Talk to dataset team):

Implement training loop.

This file will be called train.py

It will call dataset.py.

**Comment by marlenemknn** [comment]
- validation (perplexity) 
- validation (accuracy, ...) 
- training loop (optimizer?)

**Comment by maximk27** [comment]
"Token indices sequence length is longer than the specified maximum sequence length for this model (1310 > 1024). Running this sequence through the model will result in indexing errors"

I get this warning message when I run the main function. Is this harmless?

**Comment by USSiamaboat** [comment]
> "Token indices sequence length is longer than the specified maximum sequence length for this model (1310 > 1024). Running this sequence through the model will result in indexing errors"
> 
> I get this warning message when I run the main function. Is this harmless?

I think that's the tokenizer warning us that GPT2 cannot take a sequence of ids and attentions longer than 1024, but we already handle this by chunking. The way the trainer is configured right now, we shouldn't ever be attempting to feed the model any sequences longer than 1024.

**Comment by USSiamaboat** [comment]
I think the compatibility issues after merging dataset and training are almost entirely resolved, so it makes sense to move off the main `dev` branch to finish up the training script in the `feat/train` branch: https://github.com/MichiganDataScienceTeam/F24-mini-copilot/tree/feat/train
